internal/testtypes/builders.go

```diff
@@ -197,57 +88,64 @@ func (f *fakeCluster) Obj() *clusterv1.Cluster {
 	namespace                                 string
 	name                                      string
 	infrastructureClusterTemplate             *unstructured.Unstructured
 	controlPlaneMetadata                      *clusterv1.ObjectMeta
 	controlPlaneTemplate                      *unstructured.Unstructured
 	controlPlaneInfrastructureMachineTemplate *unstructured.Unstructured
-	workerMachineDeploymentTemplates          map[string]fakeClusterClassMachineDeploymentTemplates
+	workerMachineDeploymentTemplates          map[string]clusterClassMachineDeploymentTemplates
 }
 
-type fakeClusterClassMachineDeploymentTemplates struct {
+type clusterClassMachineDeploymentTemplates struct {
 	clusterv1.ObjectMeta
 	infrastructureMachineTemplate *unstructured.Unstructured
 	bootstrapTemplate             *unstructured.Unstructured
 }
 
-func newFakeClusterClass(namespace, name string) *fakeClusterClass {
-	return &fakeClusterClass{
+// NewClusterClassBuilder ...
+func NewClusterClassBuilder(namespace, name string) *ClusterClassBuilder {
+	return &ClusterClassBuilder{
 		namespace: namespace,
 		name:      name,
 	}
 }
 
-func (f *fakeClusterClass) WithInfrastructureClusterTemplate(t *unstructured.Unstructured) *fakeClusterClass {
+// WithInfrastructureClusterTemplate ...
+func (f *ClusterClassBuilder) WithInfrastructureClusterTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
 	f.infrastructureClusterTemplate = t
 	return f
 }
 
-func (f *fakeClusterClass) WithControlPlaneTemplate(t *unstructured.Unstructured) *fakeClusterClass {
+// WithControlPlaneTemplate ...
+func (f *ClusterClassBuilder) WithControlPlaneTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
 	f.controlPlaneTemplate = t
 	return f
 }
 
-func (f *fakeClusterClass) WithControlPlaneMetadata(labels, annotations map[string]string) *fakeClusterClass {
+// WithControlPlaneMetadata ...
+func (f *ClusterClassBuilder) WithControlPlaneMetadata(labels, annotations map[string]string) *ClusterClassBuilder {
 	f.controlPlaneMetadata = &clusterv1.ObjectMeta{
 		Labels:      labels,
 		Annotations: annotations,
 	}
 	return f
 }
 
-func (f *fakeClusterClass) WithControlPlaneInfrastructureMachineTemplate(t *unstructured.Unstructured) *fakeClusterClass {
+// WithControlPlaneInfrastructureMachineTemplate ...
+func (f *ClusterClassBuilder) WithControlPlaneInfrastructureMachineTemplate(t *unstructured.Unstructured) *ClusterClassBuilder {
 	f.controlPlaneInfrastructureMachineTemplate = t
 	return f
 }
 
-func (f *fakeClusterClass) WithWorkerMachineDeploymentClass(class string, labels, annotations map[string]string, infrastructureMachineTemplate, bootstrapTemplate *unstructured.Unstructured) *fakeClusterClass {
+// WithWorkerMachineDeploymentClass ...
+func (f *ClusterClassBuilder) WithWorkerMachineDeploymentClass(class string, labels, annotations map[string]string, infrastructureMachineTemplate, bootstrapTemplate *unstructured.Unstructured) *ClusterClassBuilder {
```

Comment:
Wondering if we need a intermediate builder for WorkerMachineDeploymentClass to make this more expressive (this could be also another iteration)

Reply:
Yeah - the MachineDeployment builders need a bit of a rework - the testing is quite different than from the rest of the objects. I think we should do it in this iteration without changing the initial tests and then update the tests to use the builder versions in a future PR

Reply:
Done in the latest version - WithWorkerMachineDeploymentClass now looks like the rest of the Builders.